File: src/main/java/problems/Uri1171_FrequenciaDeNumeros.java
package problems;

import java.io.IOException;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;
import java.util.Scanner;

public class Uri1171_FrequenciaDeNumeros {
    public static void main(String[] args) throws IOException {
        Scanner ler = new Scanner(System.in);

        int n = ler.nextInt();
        List<NumeroRepeticoes> lista = new ArrayList<NumeroRepeticoes>();
        for (int i = 0; i < n; i++) {

            int j = ler.nextInt();
            boolean asdf = true;
            for (int k = 0; k < lista.size(); k++) {
                if (lista.get(k).getI() == j) {
                    asdf = false;
                    lista.get(k).setRepeticoes(lista.get(k).getRepeticoes() + 1);
                }
            }
            if (asdf) {
                lista.add(new NumeroRepeticoes(j));
            }

        }
        sortAutoComplete(lista);

        for (NumeroRepeticoes num : lista) {
            System.out.println(num.getI() + " aparece " + num.getRepeticoes() + " vez(es)");
        }
    }

    public static void sortAutoComplete(List<NumeroRepeticoes> preferencia2) {
        Collections.sort(preferencia2, new Comparator<NumeroRepeticoes>() {

            @Override
            public int compare(NumeroRepeticoes auto2, NumeroRepeticoes auto1) {
                return auto2.getI() - auto1.getI();
            }
        });
    }
}

class NumeroRepeticoes {
    int i;
    int repeticoes = 1;

    public NumeroRepeticoes(int i) {
        this.i = i;
    }

    public int getI() {
        return i;
    }

    public void setI(int i) {
        this.i = i;
    }

    public int getRepeticoes() {
        return repeticoes;
    }

    public void setRepeticoes(int repeticoes) {
        this.repeticoes = repeticoes;
    }
}

File: src/main/kotlin/problems/Uri1171_FrequenciaDeNumeros.kt
package problems

import java.util.Scanner

fun main(args: Array<String>) {
    val ler = Scanner(System.`in`)
    val numeroDeNumeros: Int = ler.nextInt()
    val mapa = mutableMapOf<Int, Int>()
    for (i in 0 until numeroDeNumeros) {
        val numero: Int = ler.nextInt()

        mapa[numero] = (mapa[numero] ?: 0) + 1
    }

    mapa.toSortedMap().forEach { (numero, repeticoes) ->
        println("$numero aparece $repeticoes vez(es)")
    }
}

/*
URI Online Judge | 1171
Frequência de Números
Adaptado por Neilor Tonin, URI  Brasil

Timelimit: 1
Neste problema sua tarefa será ler vários números e em seguida dizer quantas vezes cada número aparece na entrada de dados, ou seja, deve-se escrever cada um dos valores distintos que aparecem na entrada por ordem crescente de valor.

Entrada
A entrada contém apenas 1 caso de teste. A primeira linha de entrada contem um único inteiro N, que indica a quantidade de valores que serão lidos para X (1 ≤ X ≤ 2000) logo em seguida. Com certeza cada número não aparecerá mais do que 20 vezes na entrada de dados.

Saída
Imprima a saída de acordo com o exemplo fornecido abaixo, indicando quantas vezes cada um deles aparece na entrada por ordem crescente de valor.

Exemplo de Entrada	Exemplo de Saída
7
8
10
8
260
4
10
10

4 aparece 1 vez(es)
8 aparece 2 vez(es)
10 aparece 3 vez(es)
260 aparece 1 vez(es)
 */